### PR TITLE
Change rest status code for TaskCancelledException to 400

### PR DESCRIPTION
**Pull request by iverase**
Currently when a rest action receives a`TaskCancelledException` , the response is translated to a HTTP 500 error (INTERNAL_SERVER_ERROR). This in turn gets logged as a warning in the Elasticsearch logs.

Until now this has not been a problem, but with the introduction of https://github.com/elastic/elasticsearch/pull/72688, this has become more probable and in application where there is a lot of cancellation it can lead to a very noise log with lots of warning messages.

Arguably, a task cancelation should not be consider a internal error as it is more normally related to something done by the client. There is no a good mapping between HTTP errors and task cancellation but after discussing with David and Armin, we think a 400 (BAD_REQUEST) is a better code for such an error.

This in turn will make  the error only be log in DEBUG mode.I have label the issue as a bug.

### CODE REVIEW

**Review by original-brownbear — APPROVED** [review]
LGTM assuming CI passes

File: server/src/main/java/org/elasticsearch/tasks/TaskCancelledException.java [hunk 1]
```diff
@@ -24,4 +25,9 @@ public TaskCancelledException(String msg) {
     public TaskCancelledException(StreamInput in) throws IOException{
         super(in);
     }
+
+    @Override
+    public RestStatus status() {
+        return RestStatus.BAD_REQUEST;
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/tasks/TaskCancelledException.java:31]:
```suggestion
        // Tasks are typically cancelled at the request of the client, so a 4xx status code is more accurate than the default of 500 (and
        // means we don't log every cancellation at WARN level). There's no perfect match for cancellation in the available status codes,
        // but (quoting RFC 7213) 400 Bad Request "indicates that the server cannot or will not process the request due to something that is
        // perceived to be a client error" which is broad enough to be acceptable here.
        return RestStatus.BAD_REQUEST;
```

**Review by DaveCTurner — APPROVED** [review]
LGTM, I suggested a comment explaining our reasoning.

**Comment by iverase** [comment]
Thanks @dtc, comment full on target, I have added it.

**Pull request merged**